sbt-bridge/src/xsbt/CompilerClassLoader.scala

```diff
@@ -42,6 +44,14 @@ class CompilerClassLoader(urls: Array[URL], sbtLoader: ClassLoader)
 }
 
 object CompilerClassLoader {
+  /** Cache the result of `fixBridgeLoader`.
+   *
+   *  Reusing ClassLoaders is important for warm performance since otherwise the
+   *  JIT code cache for the compiler will be discarded between every call to
+   *  the sbt `compile` task.
+   */
+  private[this] val fixedLoaderCache = new mutable.WeakHashMap[ClassLoader, ClassLoader]
```

Comment:
Are you sure? From https://docs.oracle.com/javase/8/docs/api/java/util/WeakHashMap.html: "Hash table based implementation of the Map interface, with weak keys. An entry in a WeakHashMap will automatically be removed when its key is no longer in ordinary use."